bigquery/google/cloud/bigquery/client.py

```diff
@@ -354,7 +355,18 @@ def list_datasets(
         )
 
     def dataset(self, dataset_id, project=None):
-        """Construct a reference to a dataset.
+        """Deprecated: Construct a reference to a dataset.
+
+        This method is deprecated. Construct a
+        :class:`~google.cloud.bigquery.dataset.DatasetReference` using its
+        constructor or use a string where previously a reference object was
+        used.
```

Comment:
Doesn't appear to change formatting all that much:

![image](https://user-images.githubusercontent.com/247555/73110137-eb498000-3eca-11ea-9699-5da633693729.png)

Pushed the change in case it makes things a little more machine-readable or something, though.